cargo-shuttle/src/args.rs

```diff
@@ -46,6 +48,41 @@ pub struct ProjectArgs {
     pub name: Option<ProjectName>,
 }
 
+impl ProjectArgs {
+    pub fn workspace_path(&self) -> anyhow::Result<PathBuf> {
+        let path = MetadataCommand::new()
+            .current_dir(&self.working_directory)
+            .exec()
+            .context("failed to get cargo metadata")?
+            .workspace_root
+            .into();
+
+        Ok(path)
+    }
+
+    pub fn project_name(&self) -> anyhow::Result<ProjectName> {
+        let workspace_path = self.workspace_path()?;
+
+        let meta = MetadataCommand::new()
+            .current_dir(&workspace_path)
+            .exec()
+            .unwrap();
+        let package_name = if let Some(root_package) = meta.root_package() {
+            root_package.name.clone().parse()?
+        } else {
+            workspace_path
+                .file_name()
+                .context("failed to get project name from workspace path")?
+                .to_os_string()
+                .into_string()
+                .expect("to turn workspace name into path")
```

Comment:
Should this be something like this?
```suggestion
                .expect("workspace file name should be valid unicode")
```